src/data/simple_csr_source.cc

```diff
@@ -116,5 +119,143 @@ const SparsePage& SimpleCSRSource::Value() const {
   return page_;
 }
 
+/*!
+ * Please be careful that, in official specification, the only three required fields are
```

Comment:
After we add `XGDMatrixCreateFromArrayInterfaces()` to `c_api.h` header, we should move this docstring to the docstring of `XGDMatrixCreateFromArrayInterfaces()`. This is so that this docstring will show up in https://xgboost.readthedocs.io/en/latest/dev/c__api_8h.html.

Reply:
@hcho3 I'm not entirely sure about this interface yet, so it's hidden.  More PRs will come. ;-)